### PR TITLE
feat(ESSNTL-5498): Change staleness perms

### PR DESCRIPTION
**Pull request by fellipeh**
# Overview

This PR is being created to address [ESSNTL-5498](https://issues.redhat.com/browse/ESSNTL-5498).
Changing `staleness` permissions according to new workflow

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by thearifismail — COMMENTED** [review]
The code LGTM but the pipeline failing.

**Review by kruai — CHANGES_REQUESTED** [review]
We should have at least one RBAC test that validates that this approach works.

**Review by thearifismail — APPROVED** [review]

File: tests/helpers/rbac-mock-data/inv-account-staleness-read-only.json [hunk 1]
```diff
@@ -14,6 +14,10 @@
       {
           "resourceDefinitions": [],
           "permission": "inventory:staleness:read"
```
Comment (jpramos123) [inline tests/helpers/rbac-mock-data/inv-account-staleness-read-only.json:16]: Maybe we can already add the newest `"staleness:staleness:read/write"` over here

**Review by jpramos123 — APPROVED** [review]

**Comment by fellipeh** [comment]
> i.e. it fails if the user has the staleness permission but not the host one, and vice-versa.

🤔 I understood exactly to be this way, looking the Miro flow design (link on ticket).

**Review by kruai — APPROVED** [review]
optimally we'd want to check the staleness-read allowed/prohibited as well, but I won't block on that

**Pull request merged**